packages/contracts/contracts/voting/multisig/Multisig.sol

```diff
@@ -0,0 +1,432 @@
+
+        emit MinApprovalUpdated({minApprovals: _minApprovals});
+    }
+
+    /// @notice Adds new members to the address list and updates the minimum approval parameter.
+    /// @param _members The addresses of the members to be added.
+    function addAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _addAddresses(_members);
+
+        emit AddressesAdded({members: _members});
+    }
+
+    /// @notice Removes existing members from the address list. Previously, it checks if the new addresslist length at least as long as the minimum approvals parameter requires. Note that `minApprovals` is must be at least 1 so the address list cannot become empty.
+    /// @param _members The addresses of the members to be removed.
+    function removeAddresses(address[] calldata _members)
+        external
+        auth(UPDATE_MULTISIG_SETTINGS_PERMISSION_ID)
+    {
+        _removeAddresses(_members);
+
+        // Check if the new addresslist has become shorter than the current minimum number of approvals required.
+        uint256 newAddresslistLength = addresslistLength();
+        if (newAddresslistLength < minApprovals_) {
+            revert MinApprovalsOutOfBounds({limit: newAddresslistLength, actual: minApprovals_});
+        }
+
+        emit AddressesRemoved({members: _members});
+    }
+
+    /// @notice Creates a new majority voting proposal.
+    /// @param _metadata The metadata of the proposal.
+    /// @param _actions The actions that will be executed after the proposal passes.
+    /// @param _approve_ If true, the sender will approve the proposal.
+    /// @param _tryExecution If `true`, execution is tried after the vote cast. The call does not revert if early execution is not possible.
+    /// @return proposalId The ID of the proposal.
+    function createProposal(
+        bytes calldata _metadata,
+        IDAO.Action[] calldata _actions,
+        bool _approve_,
+        bool _tryExecution
+    ) external returns (uint256 proposalId) {
+        uint64 snapshotBlock = getBlockNumber64() - 1;
+
+        if (!isListedAtBlock(_msgSender(), snapshotBlock)) {
+            revert ProposalCreationForbidden(_msgSender());
+        }
+
+        proposalId = proposalCounter.current();
+
+        // Create the proposal
+        Proposal storage proposal_ = proposals[proposalId];
+
+        proposal_.open = true;
+        proposal_.parameters.snapshotBlock = snapshotBlock;
+        proposal_.parameters.minApprovals = minApprovals_;
+        proposal_.tally.addresslistLength = addresslistLengthAtBlock(snapshotBlock); // TODO https://aragonassociation.atlassian.net/browse/APP-1417
+
+        for (uint256 i; i < _actions.length; ) {
+            proposal_.actions.push(_actions[i]);
+            unchecked {
+                ++i;
+            }
+        }
+
+        _incrementProposalCount();
+
+        if (_approve_) {
+            approve(proposalId, _tryExecution);
```

Comment:
It will fail earlier already because we require the proposal creator to be listed.

Reply:
@brickpop Do we want non-multisig members to (optionally) be allowed to create proposals?

Reply:
I should check with Selim, but I think this is a setting we'd like to allow to define.

Reply:
also, why does `_approve_` have leading and trailing underscore?

Reply:
> @Michael-A-Heuer I should check with Selim, but I think this is a setting we'd like to allow to define.

Resolved since we will tackle it in https://aragonassociation.atlassian.net/browse/APP-1431